wgsl/index.bs

```diff
@@ -344,10 +339,64 @@ Examples of predeclared objects are:
 * built-in types.
 
 A declaration must not introduce a name when that identifier is
-already in scope at the start of the declaration.
-That is, shadow names are not allowed in [SHORTNAME].
+already in scope with the same end scope as another instance of that name.
+When an identifier is referenced and multiple declarations are in scope, the
+reference is resolved to the closest preceding in scope declaration.
```

Comment:
Somehow I'd hyphenate this instance of "in scope" to be "in-scope". Can fix later.

Reply:
Hm.. I guess my suggested changes below remove that use.   Alas.